### PR TITLE
Add a pass to simplify unit

### PR DESCRIPTION
**Pull request by hkapp**
Add a pass to simplify the unit usage, especially the ones created for the control flow.
Also improve the DCE to remove unused block parameters.

I am not particularly convinced by the solution used for the latter, suggestions are welcome.

### CODE REVIEW

**Comment by densh** [comment]
Needs a rebase due to refactoring done in #401 .

**Comment by hkapp** [comment]
To match with the current passes naming convention, I guess this one should be renamed to UnitSimplification ?

**Comment by densh** [comment]
@hkapp Yeah, we've been mostly using nouns for pass names so far.

File: tools/src/main/scala/scala/scalanative/optimizer/pass/DeadCodeElimination.scala [hunk 1]
```diff
@@ -32,9 +40,39 @@ class DeadCodeElimination(implicit top: Top) extends Pass {
 
       Seq(defn.copy(insts = newinsts))
   }
+
+  private def buildParamChanges(
```
Comment (densh) [inline tools/src/main/scala/scala/scalanative/optimizer/pass/DeadCodeElimination.scala]: Computing which parameters are used and not is really a task of `analysis.UseDef`. We should move the code there and integrate it well into existing use def information.

**Comment by hkapp** [comment]
@densh I couldn't get rid completely of the subpass in DCE. Do you see a solution ?

**Comment by densh** [comment]
LGTM!

**Pull request merged**